src/components/PrivateRoute/PrivateRoute.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react';
+
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { useSetRecoilState, useRecoilValue } from 'recoil';
+
+import { LogInState } from '@/home/recoil/LogInState';
+import { useResetUserInfo } from '@/hooks/useResetUserInfo';
+import { DialogState } from '@/recoil/DialogState';
+import { api } from '@/service/TokenService';
+
+interface PrivateRouteProps {
+  toPath?: string;
+  isModalOpen?: boolean;
+  modalPath?: string;
+}
+
+export const PrivateRoute = ({
+  toPath = '/login',
+  isModalOpen = false,
+  modalPath = '/',
+}: PrivateRouteProps) => {
+  const setDialog = useSetRecoilState(DialogState);
+  const isLoggedIn = useRecoilValue(LogInState);
+  const accessToken = api.getAccessToken();
+  const resetUserInfo = useResetUserInfo();
+  const location = useLocation();
+
+  useEffect(() => {
+    if (!isLoggedIn || !accessToken) {
+      resetUserInfo();
+      if (isModalOpen) {
+        setDialog({
+          open: true,
+          type: 'login',
+          redirectPath: location.state?.from == undefined ? modalPath : null,
```

Comment:
```suggestion
          redirectPath: location.state?.from || modlaPath,
```
이렇게 수정해도 동일하게 동작할 것 같네요!

Reply:
삼항연산자 킬러...👍🏻

Reply:
[224e0be](https://github.com/yourssu/Soomsil-Web/pull/248/commits/224e0be7b79461e35b8ec6d74dcf393eebbf4e77)
감사합니다!